### DISCUSSION TITLE
Adding Shared Libraries (Sharepoint)

### DISCUSSION THREAD

**Discussion by yohann69** [spencerwooo/onedrive-vercel-index#491] [Q&A]
Hi there

I managed to deployed the website and it works amazingly well !
I have a question though, I am member of shared libraries via my office subscription (e5 dev) and I wanted to know if there is a way to add shared libraries to the website in addition of my main od storage ?

**Comment by spencerwooo** [comment]
Can you elaborate? What are shared libraries? I only have a OneDrive international and a E5 dev account, so I don't have that many features available to test on my end.

**Reply by iskycc** [reply]
Based on what he described, I think he was talking about a SharePoint site.

**Reply by myl7** [reply]
E5 admin should be able to create shared library (aka site) so a E5 is enough for test.
And I would be willing to adding the shared library support. So how about to let me take the work?

**Reply by spencerwooo** [reply]
Sure, but I'm going to have little to less time to spare for this project over the next few weeks, expect a late PR review 😢

**Reply by yohann69** [reply]
Yeah SharePoint website. They can be shared among users from a same subscription. In onedrive they are displayed as Shared libraries"

**Reply by DaleHenry** [reply]
To add to this, I have shared a folder from my sharepoint site to my e5 account personal onedrive so that the folder appears in my onedrive itself.
Can this atleast be accessible from the index?

**Reply by Rigolino1** [reply]
Someone can index shared folers or shared library?

**Reply by LuisLuii** [reply]
> Someone can index shared folers or shared library?

I explored different OneDrive index project, neither did not support shared folder. So I write it by myself in other index project . But this project UI is beautiful, I am trying to fulfil again in here

**Reply by DaleHenry** [reply]
Didn't the old index based on cloudflare workers support Sharepoint sites?
I believe it did.

**Reply by Rigolino1** [reply]
> > ¿Alguien puede indexar folers compartidos o biblioteca compartida?
> 
> Exploré diferentes proyectos de índice de OneDrive, ninguno de los dos no admitía carpetas compartidas. Así que lo escribo yo mismo en otro proyecto de índice. Pero esta interfaz de usuario del proyecto es hermosa, estoy tratando de cumplir de nuevo aquí

can you help me wen you done this please? :(

**Reply by Rigolino1** [reply]
> Didn't the old index based on cloudflare workers support Sharepoint sites? I believe it did.

bro u know how deploy on cf?, help me please

**Reply by DaleHenry** [reply]
Check out [onedrive-cf-index](https://github.com/spencerwooo/onedrive-cf-index).
I don't think its supported by dev anymore. Might still work

**Reply by LuisLuii** [reply]
> > > ¿Alguien puede indexar folers compartidos o biblioteca compartida?
> 
> > 
> 
> > Exploré diferentes proyectos de índice de OneDrive, ninguno de los dos no admitía carpetas compartidas. Así que lo escribo yo mismo en otro proyecto de índice. Pero esta interfaz de usuario del proyecto es hermosa, estoy tratando de cumplir de nuevo aquí
> 
> 
> 
> can you help me wen you done this please? :(

I have done it and committed to my forked repo. I only need the access and download function for shared file, so I will not fix the bug that not related to this. Also I have using the fodi btw   


As the bug I knew is you cannot see the preview file( it is works in my computer but after deploy and idk why) and download folder. 
Btw I know alist is supported sharepoint(not shared folder), u can try it if u need this feature

**Reply by Rigolino1** [reply]
yes. i want shared drive or sharepoint, but i need one of this for save my files, 1tb its not sufficient for me :(.
i wanna search more bro
